src/strategy/serializer.ts

```diff
@@ -1,17 +1,17 @@
+/* istanbul ignore file */
+
 import { Stream } from 'stream';
-import { StrategyOptions } from '../types';
 
 export interface StrategyStream extends Stream {
 	onEnd?<T>(encode?: BufferEncoding): T;
 }
 
-export interface SerializerStrategy {
-	serialize(
-		content: Stream,
-		options?: StrategyOptions,
-	): Promise<Stream> | Stream;
-	deserialize(
-		content: Stream,
-		options?: StrategyOptions,
-	): Promise<StrategyStream> | StrategyStream;
+export type Serialized = string | ArrayBuffer | Uint8Array;
+
+export interface SerializerStrategy<I, O extends Serialized | Stream> {
```

Comment:
Notice that a SerializerStrategy can still return a stream (as the case with gzip). But by the overall default implementation streams will only be used in the chained strategies (where the only default chaining strategy is gzip)